### PR TITLE
Game Activity Toggle and SettingsStoreAPI

### PR DESCRIPTION
**Pull request by Nuckyz**

### CODE REVIEW

**Comment by Uruknara** [comment]
Please merge this we need it so bad !

**Comment by Vendicated** [comment]
I'm really confused by some parts of this PR, i think some of the changes you've made to the context menu api here have already landed in main?

File: src/plugins/apiSettingsStore.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name: "SettingsStoreAPI",
+    description: "Adds settings groups and names to the exported object to allow for Webpack searching.",
+    authors: [Devs.Nuckyz],
+
+    patches: [
+        {
+            find: '"textAndImages","renderSpoilers"',
+            replacement: [
+                {
+                    match: /INFREQUENT_USER_ACTION.+?useSetting.+?}(?<=function \i\((\i),(\i),\i,\i\).+?)/,
```
Comment (Vendicated) [inline src/plugins/apiSettingsStore.ts:32]:
```suggestion
                    match: /(?<=INFREQUENT_USER_ACTION.{0,20}),useSetting:function/,
```

File: src/plugins/apiSettingsStore.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    replace: (m, group, name) => `${m},settingsStoreApiGroup:${group},settingsStoreApiName:${name}`
```
Comment (Vendicated) [inline src/plugins/apiSettingsStore.ts:33]:
```suggestion
                    replace: ",settingsStoreApiGroup:arguments[0],settingsStoreApiName:arguments[1],useSetting:function"`
```

File: src/plugins/apiSettingsStore.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "SettingsStoreAPI",
+    description: "Adds settings groups and names to the exported object to allow for Webpack searching.",
```
Comment (Vendicated) [inline src/plugins/apiSettingsStore.ts:24]:
```suggestion
    description: "Patches Discord's SettingsStores to expose their group and name",
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Nuckyz** [comment]
I did your changes and "rebased" from main

File: src/api/SettingsStore.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    getSetting(): T;
+    /**
+     * Update the setting value
+     * @param value The new value
+     */
+    updateSetting(value: T): Promise<void>;
+    /**
+     * React hook for automatically updating components when the setting is updated
+     */
+    useSetting(): boolean;
```
Comment (Vendicated) [inline src/api/SettingsStore.ts:36]:
```suggestion
    useSetting(): boolean;
    settingsStoreApiGroup: string;
    settingsStoreApiName: string;
```

why not add these members?

File: src/plugins/gameActivityToggle/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { getSettingStoreLazy } from "@api/SettingsStore";
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Tooltip } from "@webpack/common";
+
+const ShowCurrentGame = getSettingStoreLazy<boolean>("status", "showCurrentGame");
+
+function GameActivityToggleButton() {
+    ShowCurrentGame?.useSetting();
```
Comment (Vendicated) [inline src/plugins/gameActivityToggle/index.tsx:30]:
```suggestion
    const showCurrentGame = ShowCurrentGame?.useSetting();
```

File: src/plugins/gameActivityToggle/index.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+    return (
+        <Tooltip text="Toggle Game Activity">
+            {tooltipProps => (
+                <button
+                    {...tooltipProps}
+                    className="game-activity-toggle-btn"
+                    onClick={() => ShowCurrentGame?.updateSetting(!ShowCurrentGame.getSetting())}
```
Comment (Vendicated) [inline src/plugins/gameActivityToggle/index.tsx:38]:
```suggestion
                    onClick={() => ShowCurrentGame?.updateSetting(s => !s)}
```
i think this is a thing? If so, please fix the type above

File: src/api/SettingsStore.ts (hunk 4 above)
Comment (Vendicated) [inline src/api/SettingsStore.ts:36]:
```suggestion
    useSetting(): T;
```

File: src/plugins/gameActivityToggle/index.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                >
+                    <svg
+                        width="24"
+                        height="24"
+                        viewBox="0 96 960 960"
+                    >
+                        <g fill="currentColor">
+                            <path d="M182 856q-51 0-79-35.5T82 734l42-300q9-60 53.5-99T282 296h396q60 0 104.5 39t53.5 99l42 300q7 51-21 86.5T778 856q-21 0-39-7.5T706 826l-90-90H344l-90 90q-15 15-33 22.5t-39 7.5Zm498-240q17 0 28.5-11.5T720 576q0-17-11.5-28.5T680 536q-17 0-28.5 11.5T640 576q0 17 11.5 28.5T680 616Zm-80-120q17 0 28.5-11.5T640 456q0-17-11.5-28.5T600 416q-17 0-28.5 11.5T560 456q0 17 11.5 28.5T600 496ZM310 616h60v-70h70v-60h-70v-70h-60v70h-70v60h70v70Z" />
+                            {!ShowCurrentGame?.getSetting() && <line x1="920" y1="280" x2="40" y2="880" stroke="var(--status-danger)" stroke-width="80" />}
```
Comment (Vendicated) [inline src/plugins/gameActivityToggle/index.tsx:47]:
```suggestion
                            {!showCurrentGame && <line x1="920" y1="280" x2="40" y2="880" stroke="var(--status-danger)" stroke-width="80" />}
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

**Comment by Nuckyz** [comment]
ok done

File: src/api/SettingsStore.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+     * React hook for automatically updating components when the setting is updated
+     */
+    useSetting(): T;
+    settingsStoreApiGroup: string;
+    settingsStoreApiName: string;
+}
+
+const SettingsStores: Array<Setting<any>> | undefined = proxyLazy(() => {
+    const modId = findModuleId('"textAndImages","renderSpoilers"');
+    if (modId == null) return;
```
Comment (Vendicated) [inline src/api/SettingsStore.ts:44]: maybe log here just to make it easier to debug
```suggestion
    if (modId == null) return new Logger("SettingsStoreAPI").error("Didn't find stores module");
```

File: src/plugins/gameActivityToggle/index.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+import "./style.css";
+
+import { getSettingStore } from "@api/SettingsStore";
+import ErrorBoundary from "@components/ErrorBoundary";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+import { Tooltip } from "@webpack/common";
+
+const ShowCurrentGame = getSettingStore<boolean>("status", "showCurrentGame");
```
Comment (Vendicated) [inline src/plugins/gameActivityToggle/index.tsx:27]: i think we still need getSettingsStoreLazy because this just wouldn't work, right?

**Comment by Nuckyz** [comment]
done

**Review by Vendicated — APPROVED** [review]

**Pull request merged**